fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -2309,6 +2312,20 @@ def _fsdp_instances(self) -> List["FullyShardedDataParallel"]:
         return [m for m in self.modules() if isinstance(m, FullyShardedDataParallel)]
 
     def _remove_uncollectable_params_from_optim_state_dict(self, osd: Dict) -> Dict:
+        """Return a new state dict filtering out the ones like MoE layers, which has
+        ``no_broadcast_optim_state`` flag set.
+
+        We also make rooms for the optimizer state on rank 0.
+        """
+        # In PyTorch version 1.12, Adam's `step` state changed from an into to a singleton
```

Comment:
this is one of the fix for optimizer state handling. @sshleifer

Reply:
[nit] `step` state changed from an *into*\int to

Reply:
good catch! Thanks!